test/v1/connectors/DeleteCodingOutConnectorSpec.scala

```diff
@@ -17,58 +17,57 @@
 package v1.connectors
 
 import api.connectors.ConnectorSpec
-import api.mocks.MockHttpClient
-import mocks.MockAppConfig
-import uk.gov.hmrc.http.HeaderCarrier
-import api.models.domain.Nino
+import api.models.domain.{Nino, TaxYear}
 import api.models.outcomes.ResponseWrapper
 import v1.models.request.deleteCodingOut.DeleteCodingOutParsedRequest
 
 import scala.concurrent.Future
 
 class DeleteCodingOutConnectorSpec extends ConnectorSpec {
 
-  val nino: String    = "AA111111A"
-  val taxYear: String = "2021-22"
+  "DeleteCodingOutConnector" should {
+    "return a 204 status for a success scenario" when {
+      "a valid non-TYS request is made" in new Ifs1Test with Test {
+        def taxYear: TaxYear = TaxYear.fromMtd("2019-20")
 
-  val request: DeleteCodingOutParsedRequest = DeleteCodingOutParsedRequest(
-    nino = Nino(nino),
-    taxYear = taxYear
-  )
+        val outcome = Right(ResponseWrapper(correlationId, ()))
 
-  class Test extends MockHttpClient with MockAppConfig {
+        willDelete(
+          url = s"$baseUrl/income-tax/accounts/self-assessment/collection/tax-code/$nino/${taxYear.asMtd}"
+        ).returns(Future.successful(outcome))
 
-    val connector: DeleteCodingOutConnector = new DeleteCodingOutConnector(
-      http = mockHttpClient,
-      appConfig = mockAppConfig
-    )
+        await(connector.deleteCodingOut(request)) shouldBe outcome
+      }
 
-    MockAppConfig.ifs1BaseUrl returns baseUrl
-    MockAppConfig.ifs1Token returns "ifs1-token"
-    MockAppConfig.ifs1Environment returns "ifs1-environment"
-    MockAppConfig.ifs1EnvironmentHeaders returns Some(allowedIfs1Headers)
-  }
+      "a valid TYS request is made" in new TysIfsTest with Test {
+        def taxYear: TaxYear = TaxYear.fromMtd("2023-24")
 
-  "DeleteCodingOutConnector" when {
-    ".deleteCodingOut" should {
-      "return a 204 status for a success scenario" in new Test {
         val outcome = Right(ResponseWrapper(correlationId, ()))
 
-        implicit val hc: HeaderCarrier                      = HeaderCarrier(otherHeaders = otherHeaders ++ Seq("Content-Type" -> "application/json"))
-        val requiredIfsHeadersDelete: Seq[(String, String)] = requiredIfs1Headers ++ Seq("Content-Type" -> "application/json")
-
-        MockHttpClient
-          .delete(
-            url = s"$baseUrl/income-tax/accounts/self-assessment/collection/tax-code/$nino/$taxYear",
-            config = dummyHeaderCarrierConfig,
-            requiredHeaders = requiredIfsHeadersDelete,
-            excludedHeaders = Seq("AnotherHeader" -> "HeaderValue")
-          )
-          .returns(Future.successful(outcome))
+        willDelete(
+          url = s"$baseUrl/income-tax/${taxYear.asTysDownstream}/accounts/self-assessment/collection/tax-code/$nino"
```

Comment:
can we hardcode the taxYear into the url to ensure the correct tys endpoint is being tested